hack/update-codegen.sh

```diff
@@ -34,62 +34,18 @@ echo "Generating with deepcopy-gen"
 deepcopy-gen \
   --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1 \
-  --output-package=github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1 \
-  --output-file-base=zz_generated.deepcopy
-deepcopy-gen \
-  --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/pkg/apis/cluster \
-  --output-package=github.com/karmada-io/karmada/pkg/apis/cluster \
-  --output-file-base=zz_generated.deepcopy
-deepcopy-gen \
-  --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1 \
-  --output-package=github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1 \
-  --output-file-base=zz_generated.deepcopy
-deepcopy-gen \
-  --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/pkg/apis/work/v1alpha1 \
-  --output-package=github.com/karmada-io/karmada/pkg/apis/work/v1alpha1 \
-  --output-file-base=zz_generated.deepcopy
-deepcopy-gen \
-  --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/pkg/apis/work/v1alpha2 \
-  --output-package=github.com/karmada-io/karmada/pkg/apis/work/v1alpha2 \
-  --output-file-base=zz_generated.deepcopy
-deepcopy-gen \
-  --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/pkg/apis/config/v1alpha1 \
-  --output-package=github.com/karmada-io/karmada/pkg/apis/config/v1alpha1 \
-  --output-file-base=zz_generated.deepcopy
-deepcopy-gen \
-  --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/pkg/apis/networking/v1alpha1 \
-  --output-package=github.com/karmada-io/karmada/pkg/apis/networking/v1alpha1 \
-  --output-file-base=zz_generated.deepcopy
-deepcopy-gen \
-  --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/examples/customresourceinterpreter/apis/workload/v1alpha1 \
-  --output-package=github.com/karmada-io/karmada/examples/customresourceinterpreter/apis/workload/v1alpha1 \
```

Comment:
This item has a different output path from the other items, and may require separate handling. The operator package is also in the same situation.

Reply:
The CI doesn't echo error, maybe we need to add the verify script for `examples` and `operator`.

Reply:
I removed the `--output-package` parameter as it has no effect.
The CI verification now checks pkg and examples/customresourceinterpreter/apis, but does not include operation parts. I will add them in another PR.

Reply:
> I removed the --output-package parameter as it has no effect.

I'm surprised, why no effect?

> The CI verification now checks pkg and examples/customresourceinterpreter/apis, but does not include operation parts.

What do you mean?

Reply:
> I'm surprised, why no effect?

I reviewed the code of deepcopy-gen and found that it only uses `--output-base` to specify the output path.
Passing parameters.
![image](https://github.com/karmada-io/karmada/assets/19745536/5938180d-9432-4e53-9e4a-41e236c257fd)

Write to file.
![image](https://github.com/karmada-io/karmada/assets/19745536/d6164bbc-db97-4110-80c7-141a85d32348)


related code: https://github.com/kubernetes/code-generator

The testing result:

![image](https://github.com/karmada-io/karmada/assets/19745536/bc92adb2-882d-48ad-a3c0-2cc5c6a086da)

Reply:
> What do you mean?

We only verify directory `/pkg` and `/examples/customresourceinterpreter/apis`
https://github.com/karmada-io/karmada/blob/bb1d2e8f6feaf8881931129ef4abf8e573a9ccd4/hack/verify-codegen.sh#L9


But there is https://github.com/karmada-io/karmada/blob/master/operator/pkg/apis/operator/v1alpha1/zz_generated.deepcopy.go

We forgot to verify this file.

Reply:
Can we first fix this issue? 

I need some time to investigate why `code-generater` does not support the `--output-base`, and if we have a chance to use it in the future.

Reply:
Knative also does not utilize this parameter: https://github.com/knative/serving/blob/936ecbdb547d896d1ef260c17edb4131cfd1d2ae/hack/update-codegen.sh#L83